src/libstd/sys/unix/process.rs

```diff
@@ -254,6 +259,81 @@ impl Process {
             unsafe { libc::_exit(1) }
         }
 
+        fn walk_open_fd_brute<F>(action_fd: F) where F: Fn(RawFd) {
+            // The `getdtablesize(3)` and `sysconf(_SC_OPEN_MAX)` could miss file descriptors if
+            // the RLIMIT_NOFILE is lowered (cf. #12148) but there is no good way to list them all
+            // except with procfs (cf. `walk_open_fd_proc()`).
+            // However, an unprivileged process (i.e. without CAP_SYS_RESOURCE on Linux) is not
+            // allowed to use a file descriptor upper or equal to it's hard limit nor to change
+            // this limit upwards.
```

Comment:
Right, my tests was misleading, any process can use a file descriptor upper or equal to it's hard limit.

Do you know a way to get the full list of all usable file descriptors?